### PR TITLE
implement gethostbyaddr

### PR DESCRIPTION
**Pull request by Megazig**

### CODE REVIEW

File: libctru/source/services/soc/soc_gethostbyaddr.c [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#include "soc_common.h"
+#include <netdb.h>
+
+#define MAX_HOSTENT_RESULTS 16
+static struct hostent SOC_hostent;
+static char           *SOC_hostent_results[MAX_HOSTENT_RESULTS+1];
+static char           *SOC_hostent_alias = NULL;
+
+int h_errno = 0;
```
Comment (fincs) [inline libctru/source/services/soc/soc_gethostbyaddr.c]: `h_errno` is already defined in [soc_gethostbyname](https://github.com/Megazig/ctrulib/blob/master/libctru/source/services/soc/soc_gethostbyname.c#L9). Pinging @mtheall to see how can this be fixed.

**Comment by Megazig** [comment]
as netdb.h is included by soc_gethostbyaddr.c and int h_errno is declared in soc_gethostbyname.c, it could either be removed from soc_gethostbyaddr.c or moved to a better location that is shared between the two

**Comment by fincs** [comment]
Seems sensible.

**Comment by Megazig** [comment]
I will create a commit moving the h_errno to soc_common.c and then redo this commit without h_errno declared here.

**Pull request closed without merging**